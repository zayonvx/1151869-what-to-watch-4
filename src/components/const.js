export const onTitleClick = () => {};
export const onCardHover = () => {};
export const VIDEO = {
  delay: 1000,
  height: 175,
  width: 280,
  isMute: true,
};
