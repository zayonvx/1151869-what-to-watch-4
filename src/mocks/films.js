export const moviesData = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 3,
    title: `Macbeth`,
    image: `img/macbeth.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 4,
    title: `Aviator`,
    image: `img/aviator.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    image: `img/what-we-do-in-the-shadows.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 7,
    title: `Revenant`,
    image: `img/revenant.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 8,
    title: `Johnny English`,
    image: `img/johnny-english.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
];
