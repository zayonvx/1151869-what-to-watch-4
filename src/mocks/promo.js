export const promoData = {
  title: `Shutter Island`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  posterAlt: `The Grand Budapest Hotel poster`,
  genre: `Drama`,
  releaseDate: 2010,
};
