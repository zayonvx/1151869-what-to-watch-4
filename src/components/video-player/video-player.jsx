import React, {createRef, PureComponent} from 'react';
import PropTypes from 'prop-types';
import {VIDEO} from '../const';
import {movieTypes} from '../types';

export class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);

    this._videoRef = createRef();

    this._timeout = null;
  }


  componentDidUpdate() {
    const video = this._videoRef.current;

    if (this.props.isPlaying) {
      this._timeout = setTimeout(() => video.play(), VIDEO.delay);
    } else {
      clearTimeout(this._timeout);
      video.load();
    }
  }

  render() {
    const {movie, width, height} = this.props;
    return (
      <div className="small-movie-card__image">
        <video
          src={movie.preview}
          poster={movie.image}
          width={width}
          height={height}
          ref={this._videoRef}
          muted={this.props.isMuted}
        />
      </div>

    );
  }
}

VideoPlayer.propTypes = {
  movie: movieTypes.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  isMuted: PropTypes.bool.isRequired,
};
