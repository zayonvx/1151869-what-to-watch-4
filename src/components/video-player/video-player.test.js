import React from 'react';
import renderer from "react-test-renderer";
import {VideoPlayer} from "./video-player";
import {VIDEO} from "../const";
import {moviesData} from "../../mocks/films";

it(`VideoPlayer is renderer correctly`, () => {
  const movie = moviesData[0];
  const isPlaying = true;

  const tree = renderer.create(
      <VideoPlayer
        movie={movie}
        isPlaying={isPlaying}
        isMuted={VIDEO.isMute}
        width={VIDEO.width}
        height={VIDEO.height}
      />
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
