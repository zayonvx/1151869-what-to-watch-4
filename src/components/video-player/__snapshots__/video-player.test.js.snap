// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer is renderer correctly 1`] = `
<div
  className="small-movie-card__image"
>
  <video
    height={175}
    muted={true}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    width={280}
  />
</div>
`;
